src/applications/disability-benefits/all-claims/utils.jsx

```diff
@@ -158,6 +160,45 @@ export function prefillTransformer(pages, formData, metadata) {
   };
 }
 
+/**
+ * Transforms the related disabilities object into an array of strings
+ * @param {Object} object - The object with dynamically generated property names
+ * @return {Array} - An array of the property names with truthy values
+ */
+export function transformRelatedDisabilities(object) {
+  return Object.keys(object).filter(key => object[key]);
+}
+
+export function transform(formConfig, form) {
+  // Have to do this first or it messes up the results from transformRelatedDisabilities for some reason.
+  // The transformForSubmit's JSON.stringify transformer doesn't remove deeply empty objects, so we call
+  //  it here to remove reservesNationalGuardService if it's deeply empty.
+  const clonedData = removeDeeplyEmptyObjects(
+    JSON.parse(transformForSubmit(formConfig, form)),
+  );
+
+  // Transform the related disabilities lists into an array of strings
+  if (clonedData.vaTreatmentFacilities) {
+    const newVAFacilities = clonedData.vaTreatmentFacilities.map(facility =>
+      _.set(
+        'relatedDisabilities',
+        transformRelatedDisabilities(facility.relatedDisabilities),
+        facility,
+      ),
+    );
+    clonedData.vaTreatmentFacilities = newVAFacilities;
+  }
+  if (clonedData['view:isPOW']) {
+    clonedData['view:isPOW'] = transformRelatedDisabilities(
```

Comment:
Nice, at least there are two places where this is useful!

Reply:
Will this ever be true? `clonedData` has already been run through `transformForSubmit`, which removes `view:` properties, right?

If I do a test submit with a POW disability, I get:
```{"confinements":[{"from":"1980-05-05","to":"1999-01-01"}],"powDisabilities":{"ASHD":true}```

Reply:
Also, I'm not sure if we need to include `powDisabilities` as a child of confinements (i.e. if a `powDisabilities` list has to be associated with its given confinement period) - but I'd think so. @smluthi ^

Reply:
Oh, good point about removing the `view:` field. I had this check before I moved the `transformForSubmit` up to the top. Lemme look into it...

The latest EVSS swagger doc has 
![image](https://user-images.githubusercontent.com/12970166/48502513-91dcc000-e7f4-11e8-94d8-a6689123494b.png)

:thinking: I'm...not actually sure what the `powDisabilities` matches up to in that doc... :confused: